core_v2/sources/v2_domains.move

```diff
@@ -0,0 +1,1318 @@
+        account::get_signer_capability_address(&borrow_global<CollectionCapability>(@aptos_names_v2).capability)
+    }
+
+    fun get_token_signer(): signer acquires CollectionCapability {
+        account::create_signer_with_capability(&borrow_global<CollectionCapability>(@aptos_names_v2).capability)
+    }
+
+    inline fun get_token_addr_inline(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): address acquires CollectionCapability {
+        token::create_token_address(
+            &get_token_signer_address(),
+            &get_collection_name(is_subdomain(subdomain_name)),
+            &v2_token_helper::get_fully_qualified_domain_name(subdomain_name, domain_name),
+        )
+    }
+
+    public fun get_token_addr(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): address acquires CollectionCapability {
+        token::create_token_address(
+            &get_token_signer_address(),
+            &get_collection_name(is_subdomain(subdomain_name)),
+            &v2_token_helper::get_fully_qualified_domain_name(subdomain_name, domain_name),
+        )
+    }
+
+    fun get_record_obj(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): Object<NameRecord> acquires CollectionCapability {
+        object::address_to_object(get_token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun get_record(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): &NameRecord acquires CollectionCapability, NameRecord {
+        borrow_global<NameRecord>(get_token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun get_record_mut(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): &mut NameRecord acquires CollectionCapability, NameRecord {
+        borrow_global_mut<NameRecord>(get_token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun extract_subdomain_name(token_addr: address): Option<String> {
+        if (exists<SubdomainExt>(token_addr)) {
+            let subdomain_ext = borrow_global<SubdomainExt>(token_addr);
+            option::some(subdomain_ext.subdomain_name)
+        } else {
+            option::none<String>()
+        }
+    }
+
+    inline fun get_collection_name(is_subdomain: bool): String {
+        if (is_subdomain) {
+            v2_config::subdomain_collection_name()
+        } else {
+            v2_config::domain_collection_name()
+        }
+    }
+
+    fun validate_registration_duration(
+        registration_duration_secs: u64,
+    ) {
+        assert!(
+            registration_duration_secs % SECONDS_PER_YEAR == 0,
+            error::invalid_argument(EDURATION_MUST_BE_WHOLE_YEARS)
+        );
+
+        assert!(
+            registration_duration_secs <= v2_config::max_number_of_seconds_registered(),
+            error::out_of_range(EINVALID_NUMBER_YEARS)
+        );
+    }
+
+
+    fun validate_subdomain_expiration_policy(
```

Comment:
Do we plan to allow more types of policies? I actually can't think of any other possible policy beside these 2 so would it be simpler to just have a single auto_renew bool instead?

Reply:
fair point, Kevin, but it isn't really auto-renew, it is a distinct policy...